gems/aws-sdk-core/lib/seahorse/client/logging/formatter.rb

```diff
@@ -230,6 +231,14 @@ def summarize_string str
           end
         end
 
+        # Given an open file, this method returns a summarized inspecton
+        # string that includes the file size.
+        # @param [IO] file
+        # @return [String]
+        def summarize_file_io io
+          "#<File:#{io.path} (#{io.size} bytes)>"
```

Comment:
This looks like we took a bit of a shortcut, knowing we could list the type as `File` because of the switch condition. If we're going to also log `Tempfile`, which makes sense, we should have that display here. For example, we could accept the class as a param:

```ruby
when Tempfile then summarize_file_io(value, "Tempfile")
when File     then summarize_file_io(value, "File")
```

```ruby
def summarize_file_io(io, classname)
  "#<#{classname}:#{io.path} (#{io.size} bytes)>"
end
```